src/common/btree.c

```diff
@@ -845,8 +851,14 @@ btr_root_start(struct btr_context *tcx, struct btr_record *rec)
 	rec_dst = btr_node_rec_at(tcx, nd_off, 0);
 	btr_rec_copy(tcx, rec_dst, rec, 1);
 
-	if (btr_has_tx(tcx))
-		btr_root_tx_add(tcx); /* XXX check error */
+	if (btr_has_tx(tcx)) {
+		rc = btr_root_tx_add(tcx);
+		if (rc != 0) {
+			D_ERROR("Failed to add root into TX: %s\n",
+				strerror(errno));
+			return rc;
+		}
```

Comment:
In the volatile memory (i.e., DRAM) case, it looks like we would need to roll back the allocation done above by btr_node_alloc.

Reply:
Indeed, I'm not sure if there are other places missing rollback for vmem tree, and not sure if it's easy to rollback for vmem tree for every failure cases.  @jolivier23 any idea? should we open a new ticket to scrutinize missing rollback for vmem tree case?

Reply:
umem_tx_add_ptr returns 0 for vmem so this should never happen.